evennia/contrib/game_systems/buffs/buff.py

```diff
@@ -0,0 +1,1100 @@
+        if not context:
+            context = {}
+        self.handler.unpause(self.buffkey, context)
+
+    def reset(self):
+        """Resets the buff start time as though it were just applied; functionally identical to a refresh"""
+        self.handler.buffcache[self.buffkey]["start"] = time.time()
+
+    # endregion
+
+    # region hook methods
+    def at_apply(self, *args, **kwargs):
+        """Hook function to run when this buff is applied to an object."""
+        pass
+
+    def at_remove(self, *args, **kwargs):
+        """Hook function to run when this buff is removed from an object."""
+        pass
+
+    def at_dispel(self, *args, **kwargs):
+        """Hook function to run when this buff is dispelled from an object (removed by someone other than the buff holder)."""
+        pass
+
+    def at_expire(self, *args, **kwargs):
+        """Hook function to run when this buff expires from an object."""
+        pass
+
+    def at_pre_check(self, *args, **kwargs):
+        """Hook function to run before this buff's modifiers are checked."""
+        pass
+
+    def at_post_check(self, *args, **kwargs):
+        """Hook function to run after this buff's mods are checked."""
+        pass
+
+    def at_trigger(self, trigger: str, *args, **kwargs):
+        """Hook for the code you want to run whenever the effect is triggered.
+        Passes the trigger string to the function, so you can have multiple
+        triggers on one buff."""
+        pass
+
+    def at_tick(self, initial: bool, *args, **kwargs):
+        """Hook for actions that occur per-tick, a designer-set sub-duration.
+        `initial` tells you if it's the first tick that happens (when a buff is applied)."""
+        pass
+
+    def at_pause(self, *args, **kwargs):
+        """Hook for when this buff is paused"""
+        pass
+
+    def at_unpause(self, *args, **kwargs):
+        """Hook for when this buff is unpaused."""
+        pass
+
+    # endregion
+
+
+class Mod:
+    """A single stat mod object. One buff or trait can hold multiple mods, for the same or different stats."""
+
+    stat = "null"  # The stat string that is checked to see if this mod should be applied
+    value = 0  # Buff's value
+    perstack = 0  # How much additional value is added to the buff per stack
+    modifier = "add"  # The modifier the buff applies. 'add' or 'mult'
+
+    def __init__(self, stat: str, modifier: str, value, perstack=0.0) -> None:
+        """
+        Args:
+            stat:       The stat the buff affects. Normally matches the object attribute name
+            mod:        The modifier the buff applies. "add" for add/sub or "mult" for mult/div
+            value:      The value of the modifier
+            perstack:   How much is added to the base, per stack (including first)."""
+        self.stat = stat
+        self.modifier = modifier
+        self.value = value
+        self.perstack = perstack
+
+
+class BuffHandler(object):
```

Comment:
As a note, in Python3 you don't need to add `(object)` anymore; it's a remnant from Py2. Just `BuffHandler:` is enough.